apps/zbugs/shared/queries.ts

```diff
@@ -41,19 +42,51 @@ const issueRowSort = z.object({
 
 type IssueRowSort = z.infer<typeof issueRowSort>;
 
+const defineAuthQuery = <
```

Comment:
Turned out that adding this local helper made things much cleaner.

Without it one has to type the context and the args parameters.

One thing I was hoping was that one would be able to write:

```ts
defineQuery<AuthData>(...)
```

...but that does not work. If you provide a type parameter you have to provide all of them. You can make the rest of them optional but then we do not get type inference.

https://github.com/microsoft/TypeScript/issues/26242